Jenkinsfile

```diff
@@ -0,0 +1,200 @@
+/*
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+def setupGradle() {
+  // Delete gradle cache to workaround cache corruption bugs, see KAFKA-3167
+  dir('.gradle') {
+    deleteDir()
+  }
+  sh './gradlew -version'
+}
+
+def doValidation() {
+  try {
+    sh '''
+      ./gradlew -PscalaVersion=$SCALA_VERSION clean compileJava compileScala compileTestJava compileTestScala \
+          spotlessScalaCheck checkstyleMain checkstyleTest spotbugsMain rat \
+          --profile --no-daemon --continue -PxmlSpotBugsReport=true \"$@\"
+    '''
+  } catch(err) {
+    error('Validation checks failed, aborting this build')
+  }
+}
+
+def doTest() {
+  try {
+    sh '''
+      ./gradlew -PscalaVersion=$SCALA_VERSION unitTest integrationTest \
+          --profile --no-daemon --continue -PtestLoggingEvents=started,passed,skipped,failed "$@"
+    '''
+  } catch(err) {
+    echo 'Some tests failed, marking this build UNSTABLE'
+    currentBuild.result = 'UNSTABLE'
```

Comment:
I think this may mark a build as UNSTABLE if one of gradle's executors exited. We're trying to achieve the property that the build won't fail completely if one of the tests fails, right? If so, I think the best way is to supply `-PignoreFailures=true` at https://github.com/apache/kafka/pull/9226/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R44, and add:

```
  task integrationTest(type: Test, dependsOn: compileJava) {                                                                                                                                                                         
    ignoreFailures = userIgnoreFailures                                                                                                                                                                                              
```

```
                                                                                                                                                                                                                                     
  task unitTest(type: Test, dependsOn: compileJava) {                                                                                                                                                                                
    ignoreFailures = userIgnoreFailures                                                                                                                                                                                              
```

```
ext {                                                                                                                                                                                                                                
  userIgnoreFailures = project.hasProperty('ignoreFailures') ? ignoreFailures : false                                                                                                                                                
```

to build.gradle.

Reply:
I see, so in this case we'll let the `junit` directive set the build to unstable?

Reply:
Correct, it'll be picked up when the test output files are picked up.

Reply:
So, this seems to have caused us to lose which build had which results in the summary. 

Before we had:
![image](https://user-images.githubusercontent.com/55116/91776201-e2d0ca80-ebba-11ea-92f7-77932dbac763.png)

Now we have: 
![image](https://user-images.githubusercontent.com/55116/91776184-d9dff900-ebba-11ea-90a3-8daf3d6c384c.png)

I'll try moving the `junit` call inside the actual stage and see if that helps

Reply:
Moving the junit into the stage seemed to work